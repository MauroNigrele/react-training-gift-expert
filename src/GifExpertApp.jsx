
import { useState } from "react";
// import { AddCategory } from "./components/AddCategory";
// import { GifGrid } from "./components/GifGrid";

import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['es tal cual']);

    const onAddCategory = (newCategory) => {
        if( categories.includes(newCategory)) return;
        setCategories( [newCategory, ...categories] );
    }

    return (
        <>
            {/* Title */}
            <h1>GifExpertApp</h1>
            {/* Input */}
            <AddCategory 
                onNewCategory={ (value) => onAddCategory(value) }
            />

            {/* List */}

            { 
                categories.map( (category) => (
                    <GifGrid 
                        key={ category } 
                        category={category} 
                    />
                ))
            }
        
            {/* Title */}


        </>
    )
}
