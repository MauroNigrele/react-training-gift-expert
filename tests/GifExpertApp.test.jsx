import { render } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp"

describe('GifExpertApp Tests', () => { 
    test('should match with snapshot', () => {     
        const {container} = render( <GifExpertApp /> );
        expect(container).toMatchSnapshot();
    });
 })