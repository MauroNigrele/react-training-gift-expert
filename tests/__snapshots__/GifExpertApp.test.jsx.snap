// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Tests should match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Gifs Search"
      type="text"
      value=""
    />
  </form>
  <h3>
    anderson paak
  </h3>
  <h2>
    Loading . . .
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
